### ISSUE TITLE
I cannot find my installed extensions

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#52442]
I must have disabled gitlens a long time ago. In the meantime, I uninstalled it.

Now, I installed gitlens, and after a reload I could not see it:
<img width="300" src="https://user-images.githubusercontent.com/5047891/41653279-dcaad758-7485-11e8-80a8-68b22066eb70.png" />

I had to ask @sandy081 for help, and he showed me that there are different views and that I had to choose the installed view:

<img width="300" src="https://user-images.githubusercontent.com/5047891/41653309-f3312298-7485-11e8-892a-448124841ca2.png" />

Now I can see the gitlens extension (which I just installed), but look, there is all kind of duplication. 

I believe this viewlet should get some UX love, as I find this interaction confusing and awkward.

**Comment by alexandrudima** [comment]
i.e. 
* questionable choice to have Enabled shown by default.
* what is the use case to see both Enabled and Installed at the same time. Why would I ever want that duplication given I cannot drag and drop between the views or do anything useful with both of them at the same time.

**Comment by sandy081** [comment]
@alexandrudima First of all when an extension is installed, it should not go into disabled bucket. This looks to me the root cause of this issue.

I see your point of duplicates, I think we need good defaults here. Since some users want only enabled extensions to be seen as default, how about showing enabled, disabled and recommendation views and users can hide disabled extensions view?

**Comment by alexandrudima** [comment]
Yeah, I think when uninstalling an extension, the disabled metadata associated to that extension id should be cleared, such that when installing the same extension again it is not installed as disabled.

I think Enabled and Installed should not be sibling subviews, as there is no reasonable use-case to have them both rendered at the same time. I think when Enabled extensions are shown, Installed extensions should be hidden and vice-versa.

**Comment by ramya-rao-a** [comment]
> I think Enabled and Installed should not be sibling subviews

They won't be sibling views by default. The default at the moment is
- Installed and Recommended (in Stable)
- Enabled and Recommended (in Insiders)

We can make the default to be 
- Disabled (collapsed), Enabled and Recommended

This resolves the issue some users had with the disabled and enabled extensions mixed together in the Installed view.

If any user did like the Installed view, they can bring it back. But when doing so, us trying to hide the Enabled view feels like an overkill. The user can do that themselves.

**Comment by ramya-rao-a** [comment]
https://github.com/Microsoft/vscode/commit/c5c7d43b65bcbbb3851fbd972fad077631c707e3 updates the default view to have
- Disabled (collapsed), Enabled and Recommended

@sandy081 Do you want to keep this issue open to track why the extension ended up in the disabled state after install?

**Comment by alexandrudima** [comment]
@ramya-rao-a If the UX meeting reached consensus that's the best default, then that's great...

But my second question remains unanswered. Which of our users need to see each extension rendered two times. Why is it possible to configure such a view? i.e. Enabled/Disabled should be filters applied on the Installed extensions, they should not be separate views, as there is 0 value IMHO to have them painted at the same time:
<img width="272" alt="screen shot 2018-06-21 at 09 30 55" src="https://user-images.githubusercontent.com/5047891/41704221-f5e57efc-7535-11e8-8b02-6f51f8b929c7.png">

**Comment by sandy081** [comment]
@ramya-rao-a I will take that as a separate issue.

**Comment by ramya-rao-a** [comment]
@alexandrudima 

We have 3 groups of users who want in their default view of the Extensions viewlet to see
- Installed (both enabled/disabled in the same section/sash) 
- Enabled 
- Enabled and Disabled but in different sections/sash

If somebody chooses to see both enabled/disabled and installed, then yes, like you said there would be duplicates. But this would be something that they would explicitly choose to do so. 

If its trivial to avoid such a choice, then I agree we should guard against such choices. We should hide the Enabled/Disabled section when the Installed section is made visible by the user

But I don't believe its trivial... 
@sandy081 If we were to hide the Enabled/Disabled sections when the Installed section is made visible by the user in the default view, what would be the right place to do this? 

The `ExtensionsViewlet` only registers the views and sets defaults. Since `Installed` should still be an option in the context menu, we cannot update the `when` on the ViewDescriptor to hide it.

The visibility of each section is controlled at higher level and abstraction and is agnostic to if the viewlet is for Extensions or the Explorer. So thats not a good place either.

**Comment by ramya-rao-a** [comment]
We could go with not allowing `Installed` section at all in the default view

Folks who want a complete list of all their installed extensions can use the `Show Installed Extensions` command.

**Comment by alexandrudima** [comment]
@ramya-rao-a I have cleared local storage and that leads to the viewlet being reset to its defaults. 

#### The current default view
Issues with the current default view:
* Disabled is rendered before Enabled
* Disabled is rendered even with 0 entries
* Recommended is rendered even with 0 entries
<img width="200" src="https://user-images.githubusercontent.com/5047891/41763505-b61547b6-75fe-11e8-892b-eb628a512e6c.png" />

#### Proposed default view
I suggest that the default view be modified to look like this (i.e. have no unnecessary sub-views) -- this looks cropped because I simply deleted those dom nodes:
<img width="200" src="https://user-images.githubusercontent.com/5047891/41763620-fee93916-75fe-11e8-9a08-2e54f5f2c0b6.png" />

#### With 1 or more disabled extensions (with grouping)
Once there is at least one disabled extensions, a split can be introduced into Enabled and Disabled
<img width="200" src="https://user-images.githubusercontent.com/5047891/41763764-60489058-75ff-11e8-82e3-4a0346279edf.png" />

#### With 1 or more disabled extensions (without grouping)
This is the same as above, but with the setting turned on or off to group or not to group enabled/disabled extensions. Notice again that this looks like the default view, with no unnecessary nodes or labels.
<img width="200" src="https://user-images.githubusercontent.com/5047891/41763876-bc2233a2-75ff-11e8-8dba-ef73e947b20b.png" />

#### Also with recommendations, only paint the node if necessary (>0 recommendations).
<img width="200" src="https://user-images.githubusercontent.com/5047891/41764035-363ca834-7600-11e8-8558-54ca7fc225e2.png" />

Also, please allow me to dismiss recommendations. Once I looked at a recommended extension, I would like to consider it, and then install it or dismiss it, such that I can return to a view which does not show recommendations.

I agree you can achieve 90% of what I describe with this context menu:
<img width="200" src="https://user-images.githubusercontent.com/5047891/41764143-854f9904-7600-11e8-86ea-21fcb162e02c.png" />

Did I know that context menu existed before @sandy081 showing it to me? **No**.
Do I find that context menu confusing? **Yes**.
Did I miss-use the context menu and ended up with duplicated extensions? **Yes**.
Do I know how to get out of the state of duplicated extensions? **Yes**.
Was this a good experience for me? **Not at all**

**Comment by ramya-rao-a** [comment]
> Disabled is rendered before Enabled

I've pushed a change to change the default to
- Enabled
- Recommended
- Disabled (collapsed by default unless user decides to expand it)

> Did I miss-use the context menu and ended up with duplicated extensions? Yes.

`Installed` is no longer an option to add to the default view. We will ship this way and keep a look out for user feedback. One can always see the combined list via the `Extensions: Show Installed Extensions`

> Also, please allow me to dismiss recommendations

Individual recommendations can be dismissed in the latest Insiders. Click on the extension. On the editor that opens with extension details, there is an `Ignore Recommendation` button. This will remove the recommendation.

One can always collapse/hide the recommendation view if the whole view needs to be dismissed.

> Not showing the sub view when it has 0 extensions

This is currently not feasible to do. Whether a panel is shown or not is decided at the Viewlet level. The decision is based on the search text in the search box and not the contents of the panel/sub-view

The individual views do have the freedom to collapse themselves when there are no entries. But doing so will override user choice. So if user collapses the recommended view, it will keep getting expanded if there are recommendations. This is because the view itself doesn't know if the view was previously collapsed due to user choice or its own previous decision.

@sandy081 can attest to the above, we just discussed it this morning.
